solr/core/src/java/org/apache/solr/client/solrj/embedded/EmbeddedSolrServer.java

```diff
@@ -274,6 +273,11 @@ public void writeResults(ResultContext ctx, JavaBinCodec codec) throws IOExcepti
         req.close();
         SolrRequestInfo.clearRequestInfo();
       }
+      // NOTE: `req` must be closed _before_ core, because `req` can hold a reference to
+      // SolrIndexSearcher which can in some cases block `core.close()` from completing
+      // successfully (by holding a reference to a core Directory). Hence we cannot trivially
+      // use try-with-resources to close `core`.
+      core.close();
```

Comment:
same comment as below: couldn't `req = _parser.buildRequestFrom(core, params, getContentStreams(request));` be a try w/ resources instead of trying to do that outside with solrcore?

Reply:
Yes I agree; I think we should embrace try-with-resources as it provides structure.